linode/manager
# change: [M3-7860] - Source ACLB region info from API data and use Jakarta instead of Sydney

## Description 📝

- Updates ACLB to pull region information from API data instead of using hardcoded region data wherever possible 🗺️ 
- Updates the hardcoded beta regions to use Jakarta 🇮🇩 instead of Sydney 🇦🇺 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-03-11 at 4 15 10 PM](https://github.com/linode/manager/assets/115251059/9ce5dc25-38a6-4158-9ff3-c3ddd4619ea6) | ![Screenshot 2024-03-11 at 4 15 00 PM](https://github.com/linode/manager/assets/115251059/071c5335-eadb-41a3-bd02-463631013a6a) |
| ![Screenshot 2024-03-11 at 4 15 52 PM](https://github.com/linode/manager/assets/115251059/200683b5-ae29-4a50-8968-92c13eb57b5d) | ![Screenshot 2024-03-11 at 4 16 04 PM](https://github.com/linode/manager/assets/115251059/68c3ffbb-5832-4577-85fa-93ce323f418d) | 
| ![Screenshot 2024-03-11 at 4 17 55 PM](https://github.com/linode/manager/assets/115251059/89894399-63ba-4f34-afcf-1fa652ac5c1f) | ![Screenshot 2024-03-11 at 4 17 45 PM](https://github.com/linode/manager/assets/115251059/5c350820-bf4c-42e0-9792-097c16f5e1d4) | 
| | ⬆️ Yes this is expected. The dev environment does not have these regions so we intentionally fallback to a blank flag and region ID | 


## How to test 🧪

> [!note]
> Behavior will be different in dev compared to production because dev lacks certain regions in the API response.
> You should test this PR using both dev and production 

### Testing dev
- Login to `dev-test-aglb`
- Verify you see region IDs listed on the landing page, details page, and settings page
- Verify the create page lists the proper regions (uses Jakarta 🇮🇩 instead of Sydney 🇦🇺)

Yes, this is expected:
![Screenshot 2024-03-11 at 4 23 14 PM](https://github.com/linode/manager/assets/115251059/28cd4667-b0e6-45e6-82b8-c18d95912277)

### Testing staging/production
- Login to `prod-test-065`
- Verify you see region ID and labels on the landing page table
- Verify you see region ID and labels, and country flags on the details page and settings page
- Verify the create page lists the proper regions (uses Jakarta 🇮🇩 instead of Sydney 🇦🇺)

![Screenshot 2024-03-11 at 4 24 52 PM](https://github.com/linode/manager/assets/115251059/2d8997ea-f038-4511-8153-b4070f167d76)

## As an Author I have considered 🤔
